resources/timerbox.ts

```diff
@@ -114,23 +144,67 @@ class TimerBox extends HTMLElement {
     super();
     const root = this.attachShadow({ mode: 'open' });
     this.init(root);
-  }
 
-  // These would be used by document.registerElement, which is deprecated but
-  // ACT uses an old CEF which has this instead of the newer APIs.
-  createdCallback() {
-    const root = this.createShadowRoot();
-    this.init(root);
-  }
-  // Convert from the deprecated API names to the modern API names.
-  attachedCallback() {
-    this.connectedCallback();
-  }
-  detachedCallback() {
-    this.disconnectedCallback();
+    this._connected = false;
+
+    this.rootElement = this.shadowRoot?.getElementById('root') as HTMLDivElement;
+    this.largeBoxElement = this.shadowRoot?.getElementById('large') as HTMLDivElement;
+    this.largeBoxBackgroundElement = this.largeBoxElement.getElementsByClassName('bg')[0] as HTMLDivElement;
+    this.largeBoxForegroundElement = this.largeBoxElement.getElementsByClassName('fg')[0] as HTMLDivElement;
+    this.smallBoxElement = this.shadowRoot?.getElementById('small') as HTMLDivElement;
+    this.smallBoxBackgroundElement = this.smallBoxElement.getElementsByClassName('bg')[0] as HTMLDivElement;
+    this.smallBoxForegroundElement = this.smallBoxElement.getElementsByClassName('fg')[0] as HTMLDivElement;
+    this.timerElement = this.shadowRoot?.getElementById('timer') as HTMLDivElement;
+
+    // Constants.
+    this.kLargeSize = 50;
+    this.kSmallSize = 30;
+    this.kBorderSize = 2;
+    this.kFontSize = 16;
+    this.kAnimateMS = 100;
+
+    // Default values.
+    this._start = 0;
+    this._duration = 0;
+    this._threshold = 7;
+    this._bg = 'rgba(0, 0, 0, 0.8)';
+    this._fg = 'red';
+    this._scale = 1;
+    this._valueScale = 1;
+    this._towardTop = true;
+    this._styleFill = true;
+    this._hideAfter = -1;
+    this._bigAtZero = true;
+    this._roundUpThreshold = true;
+    this._hideTimer = 0;
+    this._timer = 0;
+    this._animationFrame = 0;
+
+    if (this.duration !== null) this._duration = Math.max(parseFloat(this.duration), 0);
+    if (this.threshold !== null) this._threshold = parseFloat(this.threshold);
+    if (this.bg !== null) this._bg = this.bg;
+    if (this.fg !== null) this._fg = this.fg;
+    if (this.scale !== null) this._scale = Math.max(parseFloat(this.scale), 0.01);
+    if (this.toward !== null) this._towardTop = this.toward !== 'bottom';
+    if (this.style !== null) this._styleFill = this.styleFill !== 'empty';
+    if (this.hideafter !== null && this.hideafter !== '') this._hideAfter = Math.max(parseFloat(this.hideafter), 0);
   }
 
-  init(root) {
+  // // These would be used by document.registerElement, which is deprecated but
```

Comment:
Yeah, I think these comments are out of date.  CEF is new enough that we don't need `registerElement` anymore.